workspaces/ui-v2/src/store/selectors/shapeSelectors.ts

```diff
@@ -0,0 +1,126 @@
+import { SerializedError } from '@reduxjs/toolkit';
+import * as Sentry from '@sentry/react';
+import sortBy from 'lodash.sortby';
+
+import {
+  AsyncStatus,
+  IShapeRenderer,
+  IFieldDetails,
+  JsonLike,
+} from '<src>/types';
+import { RootState } from '../root';
+
+// TODO write a test for this
+export const getShapeRenderer = (rootShapeId: string) => (
+  state: RootState
+): AsyncStatus<IShapeRenderer[], SerializedError> => {
+  const shapeState = state.shapes.rootShapes[rootShapeId];
+
+  if (!shapeState || shapeState.loading) {
+    return {
+      loading: true,
+    };
+  } else if (shapeState.error) {
+    return {
+      loading: false,
+      error: shapeState.error,
+    };
+  }
+
+  const resolveShape = (shapeId: string): IShapeRenderer[] => {
+    const reduxShapes = state.shapes.shapeMap[shapeId];
+    if (!reduxShapes || reduxShapes.length === 0) {
+      console.error(`Could not find shape ${shapeId} in redux store`);
+      Sentry.captureEvent({
+        message: 'Could not find shape in redux store',
+        extra: {
+          shapes: state.shapes,
+        },
+      });
```

Comment:
This might not be worth sentrying, since it's likely there's an invalid spec, but for now it's probably better to be over sensitive and mute / remove this if this becomes too noisy

Reply:
More so from the perspective that you'd expect selectors to be pure, but 🤷‍♂️